atc/api/accessor/claims_cacher.go

```diff
@@ -38,6 +40,9 @@ func NewClaimsCacher(
 }
 
 func (c *claimsCacher) GetAccessToken(rawToken string) (db.AccessToken, bool, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```

Comment:
This may be premature optimization, but I'd prefer not to hold the mutex over the database call. What do you think about using a `RWMutex` instead, something like:

```diff
diff --git a/atc/api/accessor/claims_cacher.go b/atc/api/accessor/claims_cacher.go
index 4cd964401..904d0d083 100644
--- a/atc/api/accessor/claims_cacher.go
+++ b/atc/api/accessor/claims_cacher.go
@@ -19,7 +19,7 @@ type claimsCacher struct {
 
 	cache          *lru.Cache
 	cacheSizeBytes int
-	mu             sync.Mutex // lru.Cache is not safe for concurrent access
+	mu             sync.RWMutex // lru.Cache is not safe for concurrent access
 }
 
 func NewClaimsCacher(
@@ -40,10 +40,10 @@ func NewClaimsCacher(
 }
 
 func (c *claimsCacher) GetAccessToken(rawToken string) (db.AccessToken, bool, error) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
+	c.mu.RLock()
 	claims, found := c.cache.Get(rawToken)
+	c.mu.RUnlock()
+
 	if found {
 		entry, _ := claims.(claimsCacheEntry)
 		return db.AccessToken{Token: rawToken, Claims: entry.claims}, true, nil
@@ -57,6 +57,10 @@ func (c *claimsCacher) GetAccessToken(rawToken string) (db.AccessToken, bool, er
 	if err != nil {
 		return db.AccessToken{}, false, err
 	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
 	entry := claimsCacheEntry{claims: token.Claims, size: len(payload)}
 	c.cache.Add(rawToken, entry)
 	c.cacheSizeBytes += entry.size
```

Granted, the DB call *shouldn't* be too expensive

Reply:
I did consider RWMutex but I feel like the fat Lock/defer-Unlock is significantly less likely to cause future error and I wanted favour clarity over performance.

Do you think this will actually have a significant negative performance hit? 

I'm happy to update as suggested, but fear it adds a fair bit more to be careful around.

Reply:
Fair enough - it shouldn't be a significant performance hit, anyway, since most requests won't hit the DB (so the mutex is only held for a map lookup)